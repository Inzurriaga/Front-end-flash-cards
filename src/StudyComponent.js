import React, { Component } from "react";


class Quiz extends Component {
    constructor() {
        super();
        this.state = {
            currentCardIndex: 0
        }
    }

    incrememntUp = () => {
        document.querySelector(".study-card").classList.remove("flip-back");
        if(this.state.currentCardIndex < this.props.cards.length - 1) {
            let newIndex = this.state.currentCardIndex
            newIndex++
            this.setState({ currentCardIndex: newIndex})
        }
    }

    incrememntDown = () => {
        document.querySelector(".study-card").classList.remove("flip-back");
        if(this.state.currentCardIndex >= 1) {
            let newIndex = this.state.currentCardIndex
            newIndex--
            this.setState({ currentCardIndex: newIndex})
        }
    }

    flipCard() {
        let card = document.querySelector(".study-card");
        if(!card.classList.contains("flip-back")){
            card.classList.remove("flip-front");
            card.classList.add("flip-back");
        }else{
            card.classList.remove("flip-back");
            card.classList.add("flip-front");
        }
    }
    render() {
        let index = this.state.currentCardIndex;
        return (
        <div>
            <div onClick={this.flipCard} className="study-card">
                <div className="card-front">
                    <h3>{this.props.cards[index].catagory}</h3>
                    <p>{this.props.cards[index].question}</p>
                </div>
                <div className="card-back">
                    <h3>{this.props.cards[index].catagory}</h3>
                    <p>{this.props.cards[index].answer}</p>
                </div>
            </div>
            <button onClick={this.incrememntDown}>previous</button>
            <button onClick={this.incrememntUp}>next</button>
        </div>
        )
    }
}

export default Quiz;