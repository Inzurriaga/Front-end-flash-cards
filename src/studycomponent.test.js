import React from 'react';
import StudyComponent from './StudyComponent';
import { shallow } from 'enzyme';
const increment = jest.fn()

const cards = [{"catagory":"html","question":"What are the three main tags that all HTMl structure needs?","answer":"<html>, <head>, and <body>","falseAnswer":["<header>, <section>, and <footer>","<div>, <h1>, and <p>","<section>, <div>, and <span>"],"color":"#f1652a"},{"catagory":"html","question":"How do you insert an image?","answer":"<img>","falseAnswer":["<image>","<photo>","<jpeg>"],"color":"#f1652a"},{"catagory":"html","question":"What are the basic structure for a drop down menu?","answer":"<select> and <option>","falseAnswer":["<dropdown> and <option>","<input> and <option>","<input> and <dropdown>"],"color":"#f1652a"},{"catagory":"html","question":"How do you insert a video?","answer":"<video>","falseAnswer":["<vid>","<mp3>","<vhs>"],"color":"#f1652a"}];

describe('actual study component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <StudyComponent cards={cards}/>);
  });

  it('should display study component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should invoke the toggle method when clicked', () => {
    wrapper.find('StudyComponent').forEach( (node) => {
        wrapper.find('button').simulate('click');
        expect(increment).toBeCalled();
    })
  });
});