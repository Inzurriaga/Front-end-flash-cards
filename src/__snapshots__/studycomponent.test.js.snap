// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actual study component should display study component 1`] = `
<div>
  <div
    className="study-card"
    onClick={[Function]}
  >
    <div
      className="card-front"
    >
      <h3>
        html
      </h3>
      <p>
        What are the three main tags that all HTMl structure needs?
      </p>
    </div>
    <div
      className="card-back"
    >
      <h3>
        html
      </h3>
      <p>
        &lt;html&gt;, &lt;head&gt;, and &lt;body&gt;
      </p>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    previous
  </button>
  <button
    onClick={[Function]}
  >
    next
  </button>
</div>
`;

exports[`should display all tea cards 1`] = `
<div>
  <div
    className="study-card"
    onClick={[Function]}
  >
    <div
      className="card-front"
    >
      <h3>
        html
      </h3>
      <p>
        What are the three main tags that all HTMl structure needs?
      </p>
    </div>
    <div
      className="card-back"
    >
      <h3>
        html
      </h3>
      <p>
        &lt;html&gt;, &lt;head&gt;, and &lt;body&gt;
      </p>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    previous
  </button>
  <button
    onClick={[Function]}
  >
    next
  </button>
</div>
`;

exports[`should display study component 1`] = `
<div>
  <div
    className="study-card"
    onClick={[Function]}
  >
    <div
      className="card-front"
    >
      <h3>
        html
      </h3>
      <p>
        What are the three main tags that all HTMl structure needs?
      </p>
    </div>
    <div
      className="card-back"
    >
      <h3>
        html
      </h3>
      <p>
        &lt;html&gt;, &lt;head&gt;, and &lt;body&gt;
      </p>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    previous
  </button>
  <button
    onClick={[Function]}
  >
    next
  </button>
</div>
`;
